fjrvldi6_naver_com/hardcoded-conditional.yaml

```diff
@@ -0,0 +1,19 @@
+rules:
+- id: infinite-loop
+  patterns:
+  - pattern-either:
+    - pattern: while(true) { ... }
+    - pattern: for (;;) { ... }
+  message: This loop is infinite. Make sure it terminates or has an exit condition
```

Comment:
you should add an explicit check in your rule to ensure that you aren't showing this message to an "infinite" loop that does have a break or return statement. Something like this:

https://semgrep.dev/playground/s/d6EJn